### PR TITLE
[TOS-392] Fix : disabling run button when there is only one step which is disabled

### PR DESCRIPTION
**Pull request by RagulSelvaraj**

### CODE REVIEW

File: ui/src/app/components/cases/steps-list.component.ts [hunk 1]
```diff
@@ -370,5 +372,7 @@ export class StepsListComponent extends BaseComponent implements OnInit {
 
   setTestSteps($event: TestStep[]) {
     this.testSteps = $event;
+    let EnableStep=this.testSteps?.find(testStep=> !testStep.disabled)? true: false;
```
Comment (guganabu) [inline ui/src/app/components/cases/steps-list.component.ts:375]: 1. plz use camelCase pattern for naming
2. plz recheck the enable step logic, try following, multiple steps with all disabled, one step not disabled, one step disabled, multiple step not disabled, multiple step with one disabled

File: ui/src/app/services/run-button-disable.service.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
```
Comment (guganabu) [inline ui/src/app/services/run-button-disable.service.ts:1]: does service required to find disable run button?

File: ui/src/app/components/cases/test-case-details.component.ts [hunk 3]
```diff
@@ -80,6 +83,9 @@ export class TestCaseDetailsComponent extends BaseComponent implements OnInit {
     this.testCaseService.getStepLengthEmitter().subscribe(hasInspectorFeatureres => {
       this.stepsLength = hasInspectorFeatureres;
     })
+
+    this.runButtonDisableService.enableStepdata$
```
Comment (guganabu) [inline ui/src/app/components/cases/test-case-details.component.ts:87]: does observable required here?

**Review by guganabu — CHANGES_REQUESTED** [review]

File: ui/src/app/components/cases/steps-list.component.ts (hunk 1 above)
Comment (RagulSelvaraj) [reply ui/src/app/components/cases/steps-list.component.ts]: camelCase was done...above mentioned step logic also checked and working as expected.

File: ui/src/app/services/run-button-disable.service.ts (hunk 2 above)
Comment (RagulSelvaraj) [reply ui/src/app/services/run-button-disable.service.ts:1]: yes bro needed...transfer the data from one component to another component through service

File: ui/src/app/components/cases/test-case-details.component.ts (hunk 3 above)
Comment (RagulSelvaraj) [reply ui/src/app/components/cases/test-case-details.component.ts:87]: yes required,for emitting the data.

File: ui/src/app/services/run-button-disable.service.ts (hunk 2 above)
Comment (guganabu) [reply ui/src/app/services/run-button-disable.service.ts:1]: Services should be ideally solve a contextual business solutions, run button disable is just related to state of thing which doesn't need a dedicated service.  try to move handler inside component

**Pull request closed without merging**